giskard/client/giskard_client.py

```diff
@@ -209,6 +245,10 @@ def create_project(self, project_key: str, name: str, kernel_name: str, descript
                 "name": anonymize(name),
             },
         )
+
+        if kernel_name is None:
```

Comment:
Only issue here for me is if you try to create a project already existing, you will re-create a kernel everytime.
That's why changing kernel_name to `<project_key>_<auto>` (or at least without random/uuid in name) will protect that, since kernel creation is checking that

Reply:
I updated the kernel creation to use the `project_key`. It should not create a new kernel if the project key already exists (granted it has same dependencies)